### PR TITLE
Show segment hash in segmenteditor.getAll output to aid debugging.

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

In case a user does not want to reveal a segment due to it having sensitive information, it can be hard to figure out exactly what the hash for a segment is (especially when there is an encoding or related problem). So adding it to the API output can be useful for debugging.

### Review

* [ ] Functional review done
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by sgiehl — APPROVED** [review]
Guess won't hurt to have that. Seems no tests are failing. Maybe might make sense to have a test for that API. Otherwise LGTM

**Comment by diosmosis** [comment]
Added the test. also added the extra urlencode for the definition. Will wait until the other pr is merged before merging this one.

File: plugins/SegmentEditor/tests/Integration/ApiTest.php
Comment (flamisz) [inline plugins/SegmentEditor/tests/Integration/ApiTest.php:45]: It has a typo

**Comment by flamisz** [comment]
which is the other pr you are waiting for? maybe it's merged already now?

**Comment by sgiehl** [comment]
I guess this one might meanwhile be obsolete. The hash should already be included in the API response. ping @diosmosis

**Comment by diosmosis** [comment]
Yes, unnecessary, closing this

**Pull request closed without merging**
